### DISCUSSION TITLE
Golang rewrite: Progress

### DISCUSSION THREAD

**Discussion by ejv2** [DSJAS/DSJAS#63] [General]
This discussion will contain posts detailing my progress of attempting to rewrite DSJAS in Go. This is being done on the side and has no clear deadline or end goal, but I want this rewrite to be *at least* at feature parity with mainline DSJAS. At that point, I will merge into the master branch and ship out to people.

I will make periodic posts here detailing how I am getting along.

I would like to mention first of all that DSJAS-go will not be backward compatible with old DSJAS, but I will be writing a migration tool which will allow users to migrate their code and installs to the new version. In particular, the configuration system and APIs for themes and modules will be drastically simpler, utilizing text templating instead of executing arbitrary code from theme files. This will not be as simple to migrate, but with some regular expression work, I think I could make it work. The migration tool will be worked on in parallel to DSJAS-go and will be in a separate repository under the DSJAS organization.

Thanks everybody for your kind support and patience with this being worked on.

**Comment by ejv2** [comment]
I've pushed some of the first steps in getting the site up and running. I started out by designing a system for the templating, which I hope will be much more flexible for themes and modules than we have currently. The site is not yet functional, but the tools are there for the installer to soon be ready. Once that is out of the way, we should be ready to get going and fixing everything once and for all.

**Comment by ScammerStopperZ** [comment]
Any more updates?   REALLY looking forward to being able to use this in addition to Sinister's bank environment.
One item I'd *LOVE* to see is the system automatically adding some transactions, like Sinister Spatula's Banks.

**Reply by ejv2** [reply]
Hi @ScammerStopperZ. I apologize for the radio silence on this project. As of now, for various reasons including work and other commitments, I do not have as much free time as I used to to maintain this project. I will have some time off in a couple of weeks, so I will spend at least some time on working on DSJAS again. Sadly, due to the amount of work that a full rewrite will take, it is unlikely that any such thing will happen. On the bright side, though, the existing PHP version is still there and additions are absolutely possible. I'll be making a few changes, including a fix for the problem mentioned in #69. For various other reasons (including some bad design on my part), packaged releases are also very difficult to get published properly, and so any updates will likely just be pushed to the source tree.

Again, I apologize to anybody reading this who uses this project and has been waiting for updates. It is fully my responsibility to maintain this code and I will step back up to the task when I have the opportunity.

**Reply by SinisterSpatula** [reply]
No worries, it's open source, you don't owe anyone anything.  Work on it when ever you want to. ☺️ It's very kind of you to share it.

**Comment by ejv2** [comment]
I am back working on the PHP version again. I think a complete rewrite would be counter-productive and will not be going ahead.
